File: 171.excel-sheet-column-number.java
/*
 * @lc app=leetcode id=171 lang=java
 *
 * [171] Excel Sheet Column Number
 */

// @lc code=start
class Solution {
    public int titleToNumber(String columnTitle) {
        int row = 0;
        int power = 0;
        for (int i = columnTitle.length() -1 ; i >= 0; i--) {
            int letter = columnTitle.charAt(i) - 'A' + 1;
            int value = (int) Math.pow(26,power) * letter;
            power++;
            row += value;
        }
        return row; 
    }
}
// @lc code=end

File: 171.excel-sheet-column-number.py

#
# @lc app=leetcode id=171 lang=python
#
# [171] Excel Sheet Column Number
#

# @lc code=start
class Solution(object):
    def titleToNumber(self, columnTitle):
        """
        :type columnTitle: str
        :rtype: int
        """
        row = 0
        power = 0
        for i in columnTitle[::-1]:
            value = (26**power) * (ord(i) - 64)
            row += value
            power += 1
        return row
        
        
# @lc code=end

